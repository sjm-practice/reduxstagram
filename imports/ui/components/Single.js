// @flow
import React, {
  Component,
} from "react";
import Photo from "./Photo";
import type { IPosts } from "../../state/reducers/posts";

type IProps = {
  posts: IPosts,
};

// eslint-disable-next-line react/prefer-stateless-function
class Single extends Component {
  props: IProps;

  render() {
    const i = this.props.posts.findIndex(post => post.code === this.props.params.postId);
    const post = this.props.posts[i];

    return (
      <div className="single-photo">
        <Photo index={i} post={post} {...this.props} />
      </div>
    );
  }
}

export default Single;
