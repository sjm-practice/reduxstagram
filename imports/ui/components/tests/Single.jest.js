// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import React from "react";
import renderer from "react-test-renderer";
import Single from "../Single";

describe("<Single />", function () {
  it("matches render snapshot", function () {
    const tree = renderer.create(<Single />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

