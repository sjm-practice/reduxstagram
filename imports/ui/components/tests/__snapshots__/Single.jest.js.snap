exports[`<Single /> matches render snapshot 1`] = `
<div
  className="single-photo">
  single photo
</div>
`;
